<launch>

  <rosparam param="base_width">0.3</rosparam>
  <rosparam param="ticks_meter">14865</rosparam>


  <node name="arduino_node" pkg="agv_nav" type="arduino.py" output="screen">
    <!--Write the parameters of board like port, baudRate-->
    <rosparam param="port">/dev/ttyACM0</rosparam>
    <rosparam param="baudRate">115200</rosparam>

  </node>



  <node name="left_pid_velocity" pkg="agv_nav" type="pid_velocity.py" output="screen">
      <remap from="wheel" to="lencoder"/>
      <remap from="motor_cmd" to="lwheel_speed"/>
      <remap from="wheel_vtarget" to="lwheel_vtarget"/>
      <remap from="wheel_vel" to="left_wheel_vel"/>
    
      <rosparam param="Kp">0.0276</rosparam>
      <rosparam param="Ki">0.035</rosparam>
      <rosparam param="Kd">0.0743</rosparam>
      <rosparam param="out_min">13</rosparam>
      <rosparam param="out_max">250</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">4</rosparam>
      <rosparam param="rolling_pts">5</rosparam>

  </node>

  <node name="right_pid_velocity" pkg="agv_nav" type="pid_velocity.py" output="screen">
      <remap from="wheel" to="rencoder"/>
      <remap from="motor_cmd" to="rwheel_speed"/>
      <remap from="wheel_vtarget" to="rwheel_vtarget"/>
      <remap from="wheel_vel" to="right_wheel_vel"/>
      <rosparam param="Kp">0.0276</rosparam>
      <rosparam param="Ki">0.035</rosparam>
      <rosparam param="Kd">0.0743</rosparam>
      <rosparam param="out_min">13</rosparam>
      <rosparam param="out_max">250</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">4</rosparam>
      <rosparam param="rolling_pts">5</rosparam>



 </node>


  <node pkg="agv_nav" type="twist_to_motors.py" name="twist_to_motors" output="screen"/>
  <!--node pkg="chefbot_bringup" type="diff_tf.py" name="diff_tf" output="screen"--> 


</launch>